### DISCUSSION TITLE
Implement IIncrementalGenerator

### DISCUSSION THREAD

**Discussion by devbased** [canton7/PropertyChanged.SourceGenerator#17] [Ideas] [answered]
From [dotnet/roslyn](https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md) 

## Summary 
Incremental generators are intended to be a new API that exists alongside source generators to allow users to specify generation strategies that can be applied in a high performance way by the hosting layer.

### High Level Design Goals
Allow for a finer grained approach to defining a generator
Scale source generators to support 'Roslyn/CoreCLR' scale projects in Visual Studio
Exploit caching between fine grained steps to reduce duplicate work
Support generating more items that just source texts
Exist alongside ISourceGenerator based implementations

**Comment by canton7** [comment, marked as answer]
Incremental SGs don't support the sort of analysis that we do. See https://github.com/dotnet/roslyn/discussions/60273.

When incremental SGs were introduced, they were marketed as being useful for Roslyn-scale codebase. Now they've become a bit of a fad. Do you have a performance problem with PropertyChanged.SourceGenerator which an incremental SG would solve?

**Comment by devbased** [comment]
Oh, I see that you are aware of IIncrementalGenerator.
I thought it should be used by default and didn't know it has such issues, thats interesting...

Thank you for your response!
